accepted/2020/windows-specific-apis/windows-specific-apis.md

```diff
@@ -0,0 +1,475 @@
+means the custom attribute will look as follows:
+
+```C#
+[MinimumOSPlatform("windows7.0")]
+```
+
+## Entire assemblies
+
+There is a set of assemblies that we can mark wholesale as being
+Windows-specific:
+
+> **OPEN**: Should `System.Threading.Overlapped.dll` be removed?
+> **OPEN**: I was told that part of `DirectoryServices` is now cross-platform. Is that true?
+
+* **Microsoft.Win32.Registry.dll**
+* **Microsoft.Win32.Registry.AccessControl.dll**
+* **Microsoft.Win32.SystemEvents.dll**
+* **System.Diagnostics.EventLog.dll**
+* **System.Diagnostics.PerformanceCounter.dll**
+* **System.DirectoryServices.dll**
+* **System.DirectoryServices.AccountManagement.dll**
+* **System.DirectoryServices.Protocols.dll**
+* **System.IO.FileSystem.AccessControl.dll**
+* **System.IO.FileSystem.DriveInfo.dll**
+* **System.IO.Pipes.AccessControl.dll**
+* **System.Management.dll**
+* **System.Net.Http.WinHttpHandler.dll**
+* **System.Runtime.InteropServices.WindowsRuntime.dll**
+* **System.Security.AccessControl.dll**
+* **System.Security.Cryptography.Cng.dll**
+* **System.Security.Cryptography.ProtectedData.dll**
+* **System.Security.Principal.Windows.dll**
+* **System.ServiceProcess.ServiceController.dll**
+* **System.Threading.AccessControl.dll**
+* **System.Threading.Overlapped.dll**
+
+## Specific APIs
+
+### System
+
+> **OPEN** Should we treat those as Windows-only or should we consider them as
+> partially portable?
+
+* **Console**
+    - `Beep(Int32, Int32)`
+    - `get_CapsLock()`
+    - `get_CursorVisible()`
+    - `get_NumberLock()`
+    - `get_Title()`
+    - `MoveBufferArea(Int32, Int32, Int32, Int32, Int32, Int32)`
+    - `MoveBufferArea(Int32, Int32, Int32, Int32, Int32, Int32, Char, ConsoleColor, ConsoleColor)`
+    - `set_BufferHeight(Int32)`
+    - `set_BufferWidth(Int32)`
+    - `set_CursorSize(Int32)`
+    - `set_WindowHeight(Int32)`
+    - `set_WindowLeft(Int32)`
+    - `set_WindowTop(Int32)`
+    - `set_WindowWidth(Int32)`
+    - `SetBufferSize(Int32, Int32)`
+    - `SetWindowPosition(Int32, Int32)`
+    - `SetWindowSize(Int32, Int32)`
+
+### System.Configuration
+
+> **OPEN** Should we mark the entire type?
+
+* **DpapiProtectedConfigurationProvider**
+    - `Decrypt(XmlNode)`
+    - `Encrypt(XmlNode)`
+
+### System.Diagnostics
+
+* **Process**
+    - `GetProcessById(Int32, String)`
+    - `GetProcesses(String)`
+    - `GetProcessesByName(String, String)`
+    - `set_MaxWorkingSet(IntPtr)`
+    - `set_MinWorkingSet(IntPtr)`
+    - `Start(String, String, SecureString, String)`
+    - `Start(String, String, String, SecureString, String)`
+* **ProcessStartInfo**
+    - `get_Domain()`
+    - `get_LoadUserProfile()`
+    - `get_Password()`
+    - `get_PasswordInClearText()`
+    - `set_Domain(String)`
+    - `set_LoadUserProfile(Boolean)`
+    - `set_Password(SecureString)`
+    - `set_PasswordInClearText(String)`
+* **ProcessThread**
+    - `set_PriorityLevel(ThreadPriorityLevel)`
+    - `set_ProcessorAffinity(IntPtr)`
```

Comment:
`IdealProcessor`, `ResetIdealProcessor` are nops on Unix. Should they be marked too?

Reply:
I'm on the fence for those. On the one hand, if the APIs don't throw it might be overkill. On the other hand, if the APIs don't do what I want, it's also not great. I'd say for now let's err on the side of reducing false positive and not mark those.

/cc @stephentoub

Reply:
Yeah, let's leave them unannotated.  Lots of things end up being best-effort, and the cleanest bar right now is whether the API is unusable because it throws an exception unconditionally.